lib/xrp/XrpRpc.js

```diff
@@ -0,0 +1,248 @@
+      server: connectionString
+    });
+    this.address = address;
+    this.emitter = new EventEmitter();
+  }
+
+  async asyncCall(method, args) {
+    return new Promise((resolve) => {
+      this.rpc.connect().then(() => {
+        return this.rpc[method](args).then((result) => {
+          this.rpc.disconnect().then(() => {
+            return resolve(result);
+          });
+        });
+      });
+    });
+  }
+
+  async asyncRequest(method, args) {
+    return new Promise((resolve) => {
+      this.rpc.connect().then(() => {
+        return this.rpc.request(method, args).then((result) => {
+          this.rpc.disconnect().then(() => {
+            return resolve(result);
+          });
+        });
+      });
+    });
+  }
+
+  async unlockAndSendToAddress({ address, amount, secret }) {
+    if (secret === undefined) {
+      secret = await passwordPromptAsync('> ');
+    }
+    console.warn('Unlocking for a single transaction.');
+    return await this.sendToAddress({ address, amount, secret });
+  }
+
+  async sendToAddress({ address, amount, passphrase, tag, invoiceID, secret, alreadyConnected}) {
+    return this.signTransaction({address, amount, passphrase, tag, invoiceID, secret}).then((signedTx) => {
+      return this.sendRawTransaction({ signedTx, alreadyConnected }).then((txHash) => {
+        return txHash;
+      });
+    });
+  }
+
+  async getRawTransaction({ address, amount, tag, invoiceID, sourceAddress }) {
+    const payment = {
+      source: {
+        address: sourceAddress,
+        tag: tag || undefined,
+        maxAmount: {
+          value: amount.toString(),
+          currency: this.config.currency
+        }
+      },
+      destination: {
+        address: address,
+        tag: tag || undefined,
+        amount: {
+          value: amount.toString(),
+          currency: this.config.currency
+        }
+      },
+      invoiceID: invoiceID || undefined
+    };
+    await this.rpc.connect();
+    let prepared = await this.rpc.preparePayment(sourceAddress, payment);
+    await this.rpc.disconnect();
+    return prepared;
+  }
+
+  async signTransaction({ address, amount, tag, invoiceID, secret, alreadyConnected }) {
+    if (!secret) {
+      throw new Error('Secret not provided');
+    }
+    let sourceAddress = this.rpc.deriveAddress(this.rpc.deriveKeypair(secret).publicKey);
+    const payment = {
+      source: {
+        address: sourceAddress,
+        tag: tag || undefined,
+        maxAmount: {
+          value: amount.toString(),
+          currency: this.config.currency
+        }
+      },
+      destination: {
+        address: address,
+        tag: tag || undefined,
+        amount: {
+          value: amount.toString(),
+          currency: this.config.currency
+        }
+      },
+      invoiceID: invoiceID || undefined
+    };
+    if (!alreadyConnected) {
+      await this.rpc.connect();
+    }
+    return this.rpc.preparePayment(sourceAddress, payment).then(async (prepared) => {
```

Comment:
should we use asyncCall here as well for these connecting disconnecting

Reply:
Done